### PR TITLE
Changes to validate ip and autnum paths.

### PR DESCRIPTION
**Pull request by tlm**
Validates for current state

### CODE REVIEW

File: src/main/java/net/apnic/whowas/types/Parsing.java [hunk 1]
```diff
@@ -1,29 +1,62 @@
...
+        Patterns.INTEGER
+            .next(Patterns.isChar('.'))
+            .times(3)
+            .next(Patterns.INTEGER).toScanner("IPv44 Address").source()
+            .map(s ->
+            {
                 try {
                     return new IP(Inet4Address.getByName(s));
                 } catch (UnknownHostException e) {
                     throw new RuntimeException(e);
```
Comment (GKTheOne) [inline src/main/java/net/apnic/whowas/types/Parsing.java]: {suggestion} Maybe a comment somewhere saying that the unknown host exception is only expected if a non-ip address is given to Inet4/6Address? (which the supplied pattern should prevent) Seeing the RuntimeException wrapper here threw me as I would have expected a ParseException (or similar).

File: src/test/java/net/apnic/whowas/types/ParsingTest.java [hunk 2]
```diff
@@ -13,11 +16,27 @@ public void testParseInterval() throws Exception {
         Parsing.parseInterval("2403:AD00:1000::/48");
         Parsing.parseInterval("::");
         Parsing.parseInterval("2403:ad00:1000::1");
+        Parsing.parseInterval("::2004");
+        Parsing.parseInterval("::3");
```
Comment (GKTheOne) [inline src/test/java/net/apnic/whowas/types/ParsingTest.java:54]: {suggestion} Add the edge cases? eg, for v6, "::/0", "ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff/128", and 7fff/8000 slash 0/63/64/127/128, etc?

**Review by GKTheOne — APPROVED** [review]

File: src/test/java/net/apnic/whowas/types/ParsingTest.java (hunk 2 above)
Comment (tlm) [reply src/test/java/net/apnic/whowas/types/ParsingTest.java:54]: Done & Done

File: src/main/java/net/apnic/whowas/types/Parsing.java (hunk 1 above)
Comment (tlm) [reply src/main/java/net/apnic/whowas/types/Parsing.java]: Done & Done
